lib: location: Update documentation (#19727)
nrfconnect/sdk-nrf: doc/nrf/libraries/modem/location.rst

* GNSS positioning

>   * Uses :ref:`GNSS interface<gnss_interface>` for getting the location.

Reviewer: ```suggestion
  * Uses :ref:`GNSS interface <gnss_interface>` for getting the location.
```
There should be a space here. Applies to all the references.

Reviewer 2: Works both with or without the space.

Reviewer: I was referring to https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/dev_model_and_contributions/documentation/styleguide.html#internal_links. :) Please leave it as is .

Limitations
***********

* The Location library can only have one application registered at a time.
  If there is already an application handler registered, another initialization will override the existing handler.

> * GNSS API should not be used directly by the application when using the Location library.

Reviewer 3: ```suggestion
* :ref:`nrfxlib:gnss_interface` should not be used directly by the application when using the Location library.
```

Author: I did consider using the link instead, but the title is "Global Satellite Navigation System", not "GNSS". I would rather use "GNSS API" than "Global Satellite Navigation System API". On the other hand, I'm wondering if we should change the title to "GNSS" instead. It seems that it was changed to "Global Navigation Satellite System" by this PR: https://github.com/nrfconnect/sdk-nrfxlib/commit/1e3850de8b28809feb730b805f3033d3c0bcd164

Reviewer 3: I noticed that the name of the API was changed in that PR.
I think we should refer with the link in any case. If the name of the API doesn't make sense, we should change that.

I did have same though that the change done in that PR is not good and I'd use GNSS instead.

Author: Ok, I'll update this and make an nrfxlib PR to change the name of the interface.

> * :ref:`GNSS interface<nrfxlib:gnss_interface>` should not be used directly by the application when using the Location library.

Reviewer 2: ```suggestion
* The :ref:`GNSS interface<nrfxlib:gnss_interface>` should not be used directly by the application when using the Location library.
```

>   Using the GNSS API from both the application and the Location library may lead to unexpected behavior.

Reviewer 3: And what about the dynamics mode?

So it's not required by Location library that uses single fixes. Is this clear for the user?
And is it clear that dynamics mode is not helpful for single fix mode (non-continuous)?

Author: There are a lot of bells and whistles in the GNSS API and we can't explicitly list everything here. I would rather just say, that you shouldn't mix Location library and GNSS API usage. If you do so, you do so at your own risk.

I'll discuss with Tuomas if we should add a generic comment for dynamics mode regarding the GNSS operational mode.

Reviewer 3: I'm not saying these things should be documented here, but I feel we should mention these topics somewhere.
Location library using single fixes could be in this document (I don't know where else would it be) and dynamics mode not being helpful in single fix mode could be documented into GNSS API.

Author: The dynamics mode does have some effect also in non-continuous modes. Thus, there's no need to update the dynamics mode documentation, I think.

Using GNSS API directly to configure dynamics mode with the Location lib is another thing. Personally, I would like to discourage GNSS API usage with the Location library, because that can lead to problems. Instead of providing instructions which do exactly that, we could consider adding support for configuring the dynamics mode in the Location library. @hiltunent, any opinions?

Reviewer 3: This is all a bit confusing and contradicting what we discussed last week. And I wouldn't like to add a parameter which is close to useless. But it seems we are unwilling to instruct not to use dynamics mode in non-continuous mode either...
Just leaving things as they are (unclear raising questions in user's heads) is an option but not an ideal one either as someone else will try the same mistakes. Or will try to find a way to use dynamics mode thinking it will improve their accuracy data a lot.

>   Using the GNSS API from both the application and the Location library may lead to unexpected behavior.

Author: We need to first solve the naming of the gnss_interface document. It used to have also the word "interface" in it. Now all references to it have been changed from "GNSS interface" to "Global Navigation Satellite System", which is not good. After the nrfxlib document naming has been settled, I will update this PR accordingly.

>   Using the GNSS API from both the application and the Location library may lead to unexpected behavior.

Author: It was decided that the gnss_interface title is "GNSS". Because of this I added custom link text ("GNSS interface") to all references to that page.
